### PR TITLE
Class regression

### PR DESCRIPTION
**Pull request by murphymadeleine21**

### CODE REVIEW

**Comment by murphymadeleine21** [comment]
Confusion Matrix Controller vs. Progressor: [[29 16] [14 39]] 
Confusion Matrix Viremic vs. Nonviremic: [[30 16] [14 38]] 
Our numbers are smaller because they somehow use all 181 subjects for this case–I guess not getting rid of any columns that have missing data? We only use 98 subjects here. Otherwise, not sure if this seems reasonable?

File: syserol/regression.py [hunk 1]
```diff
@@ -28,3 +28,86 @@ def patientComponents(nComp=1):
...
+    df_merged = reduce(lambda  left,right: pd.merge(left,right,on=['subject'],
+                                                    how='inner'), data_frames)
+    df_merged = df_merged.dropna()
+
+    #Subset, Z score
+    df_class = df_merged[["class.cp", "class.nv"]]
+    df_variables = df_merged.drop(['subject','class.etuv', 'class.cp', 'class.nv'], axis = 1)
+    df_variables = df_variables.apply(zscore) #Logistic Regression doesn't have normalization?
+
+    """Predict Controller vs. Progressor"""
```
Comment (aarmey) [inline syserol/regression.py]: These should just be standard comments, not doc strings.

File: syserol/regression.py [hunk 2]
```diff
@@ -28,3 +28,86 @@ def patientComponents(nComp=1):
...
+
+    Y_pred1  = cross_val_predict(LogisticRegressionCV(), X1, Y1)
+    model1 = LogisticRegressionCV().fit(X1, Y1)
+
+    print(model1.coef_)
+    accuracyCvP = confusion_matrix(Y1, Y_pred1)
+    print(f"Confusion Matrix Controller vs. Progressor: {accuracyCvP} \n")
+
+    
+    """Predict Viremic vs. Nonviremic"""
```
Comment (aarmey) [inline syserol/regression.py]: Here too.

File: syserol/regression.py [hunk 3]
```diff
@@ -28,3 +28,86 @@ def patientComponents(nComp=1):
...
+    Y2 = df_class['class.nv']
+    Y2 = (Y2 == 'viremic').astype(int) #viremic = 1, nonviremic = 0
+    X2 = df_variables 
+    Y_pred2 = np.empty(Y2.shape)
```
Comment (aarmey) [inline syserol/regression.py]: Don't need to pre-allocate `Y_pred2`

File: syserol/regression.py [hunk 4]
```diff
@@ -28,3 +28,86 @@ def patientComponents(nComp=1):
...
+    #Subset, Z score
+    df_class = df_merged[["class.cp", "class.nv"]]
+    df_variables = df_merged.drop(['subject','class.etuv', 'class.cp', 'class.nv'], axis = 1)
+    df_variables = df_variables.apply(zscore) #Logistic Regression doesn't have normalization?
+
+    """Predict Controller vs. Progressor"""
+    Y1 = df_class['class.cp']
+    Y1 = (Y1 == 'controller').astype(int) #controllers are 1s, progressors are 0s 
+    X1 = df_variables
+    Y_pred1 = np.empty(Y1.shape)
```
Comment (aarmey) [inline syserol/regression.py]: Don't need to pre-allocate `Y_pred1`

**Review by aarmey — CHANGES_REQUESTED** [review]
Do a little cleanup, but this does look right.

**Comment by aarmey** [comment]
I think they probably threw out the variables with missing data, not the subjects.

**Comment by murphymadeleine21** [comment]
Glycan only has 98 subjects, do we want to keep glycan but reduce # of subjects to 98 (what we have currently), or get rid of the glycan data to include all 181 luminex/luminex-igg data?

**Comment by aarmey** [comment]
Here, I'd like to match what they did. I'm guessing they threw out the glycans then.

**Comment by murphymadeleine21** [comment]
Confusion Matrix Controller vs. Progressor: [[53 32][24 72]] 
Confusion Matrix Viremic vs. Nonviremic: [[62 27][20 72]] 
This is pretty close to what they have.

**Review by aarmey — APPROVED** [review]

**Pull request merged**